xla/service/gpu/ir_emitter_triton.cc

```diff
@@ -744,11 +758,23 @@ absl::StatusOr<Value> EmitScope(
 // are some signs that show that this was intended to be used as an in-out
 // parameter which would give a hint to Triton which cluster dims we prefer to
 // use, but that's not the case currently.
+#ifndef TENSORFLOW_USE_ROCM
 absl::Status CreateTritonPipeline(
-    mlir::OpPassManager& pm, const se::CudaComputeCapability& cc,
+    mlir::OpPassManager& pm, const se::GpuComputeCapability& cc,
     const TritonGemmConfig& config,
     mt::nvidia_gpu::ClusterInfo& out_cluster_info) {
-  const int ccAsInt = cc.major * 10 + cc.minor;
+#else
+absl::Status CreateTritonPipeline(
+    mlir::OpPassManager& pm, const se::GpuComputeCapability& cc,
+    const TritonGemmConfig& config) {
+#endif
```

Comment:
In general I would recommend to reduce the number of ifdefs inside the code, to close to 0.
Could we have some kind of abstraction, like have a 
Something class, and a derived CudaSomething and RocmSomething and just use a runtime if based on whether the se::GpuComputeCapability is cuda or rocm?

Reply:
I talked with my colleagues and a class is not really needed.
We could use something like this:

something.h:
```
int something();
```

something_cuda.cc:
```
int something() {
 // implemented with cuda
}
```

something_rocm.cc:
```
int something() {
 // implemented with rocm
}
```
And then build either something_rocm.cc or something_cuda.cc with a select.
So ir_emitter_triton.cc would remain mostly device vendor independent and just include the shared "something.h'.

Reply:
Thank you for the proposal and clarification. I will change the implementation (based on proposal) and update PR.

Reply:
thank you!